src/connectors/voice-search/connectVoiceSearch.ts

```diff
@@ -116,27 +84,67 @@ const connectVoiceSearch: VoiceSearchConnector = function connectVoiceSearch(
           language,
           onQueryChange: query => (this as any)._refine(query),
           onStateChange: () => {
-            render({
-              isFirstRendering: false,
-              instantSearchInstance,
-              voiceSearchHelper: (this as any)._voiceSearchHelper,
-            });
+            renderFn(
+              {
+                ...this.getWidgetRenderState(initOptions),
+                instantSearchInstance,
+              },
+              false
+            );
           },
         });
 
-        render({
-          isFirstRendering: true,
-          instantSearchInstance,
-          voiceSearchHelper: (this as any)._voiceSearchHelper,
-        });
+        renderFn(
+          {
+            ...this.getWidgetRenderState(initOptions),
+            instantSearchInstance,
+          },
+          true
+        );
       },
 
-      render({ instantSearchInstance }) {
-        render({
-          isFirstRendering: false,
-          instantSearchInstance,
-          voiceSearchHelper: (this as any)._voiceSearchHelper,
-        });
+      render(renderOptions) {
+        const { instantSearchInstance } = renderOptions;
+        renderFn(
+          {
+            ...this.getWidgetRenderState(renderOptions),
+            instantSearchInstance,
+          },
+          false
+        );
+      },
+
+      getRenderState(renderState, renderOptions) {
+        return {
+          ...renderState,
+          voiceSearch: this.getWidgetRenderState(renderOptions),
+        };
+      },
+
+      getWidgetRenderState() {
+        const {
+          isBrowserSupported,
+          isListening,
+          startListening,
+          stopListening,
+          getState,
+        } = (this as any)._voiceSearchHelper;
```

Comment:
Ideally we wouldn't need this (maybe connectorState), but since the PR didn't change it, it's ok to keep as is I think.